### PR TITLE
validations and model specs

### PR DESCRIPTION
**Pull request by Zeph180**
1. For the User model:

-  Name must not be blank.
-  PostsCounter must be an integer greater than or equal to zero.
2. For the Post model:

- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

**Review by emmanuellekamwa — CHANGES_REQUESTED** [review]
**STATUS: CHANGES REQUESTED ❌⛔**
Hello @Zeph180 👋
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Descriptive PR.
- Good commit messages.
-  Correct git-flow.
- All the requirements have been implemented correctly.

## Required Changes ♻️
- Please delete the live demo section on your readme since it is not required in this milestone.

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      comment = post.comments.create(text: 'Great post!', author_id: user.id, user_id: user.id)
+      expect do
+        comment.update_comment_count
+        post.reload
+      end.to change { post.comments_count }.by(1)
+    end
+  end
+
+  describe 'associations' do
+    it { should belong_to(:post) }
```
Comment (franclobo) [inline spec/models/comment_spec.rb:18]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    it { should belong_to(:author).class_name('User') }
```
Comment (franclobo) [inline spec/models/comment_spec.rb:19]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:text) }
```
Comment (franclobo) [inline spec/models/comment_spec.rb:23]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    it { should validate_length_of(:text).is_at_most(1000) }
```
Comment (franclobo) [inline spec/models/comment_spec.rb:24]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

File: spec/models/comment_spec.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    it { should validate_presence_of(:user_id) }
```
Comment (franclobo) [inline spec/models/comment_spec.rb:25]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

File: spec/models/comment_spec.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    it { should validate_presence_of(:post_id) }
```
Comment (franclobo) [inline spec/models/comment_spec.rb:26]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

File: spec/models/comment_spec.rb [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    it { should validate_presence_of(:author_id) }
```
Comment (franclobo) [inline spec/models/comment_spec.rb:27]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

File: spec/models/user_spec.rb [hunk 8]
```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { described_class.new(name: 'John') }
+
+  describe 'associations' do
+    it { should have_many(:posts).with_foreign_key(:author_id).dependent(:destroy) }
```
Comment (franclobo) [inline spec/models/user_spec.rb:7]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

File: spec/models/user_spec.rb [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    it { should have_many(:comments).with_foreign_key(:author_id) }
```
Comment (franclobo) [inline spec/models/user_spec.rb:8]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.
image

File: spec/models/user_spec.rb [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    it { should have_many(:likes).with_foreign_key(:author_id).class_name('Like').dependent(:destroy) }
```
Comment (franclobo) [inline spec/models/user_spec.rb:9]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

File: spec/models/user_spec.rb [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:name) }
```
Comment (franclobo) [inline spec/models/user_spec.rb:13]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

File: spec/models/user_spec.rb [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+    it { should validate_numericality_of(:posts_count).only_integer.is_greater_than_or_equal_to(0) }
```
Comment (franclobo) [inline spec/models/user_spec.rb:14]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

File: spec/models/user_spec.rb [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  describe '#last_three_posts' do
+    let!(:user) { create(:user) }
```
Comment (franclobo) [inline spec/models/user_spec.rb:18]: - **[OPTIONAL]** This test returns an error, please fix it to pass them all.

image

**Review by franclobo — APPROVED** [review]
Hi @Zeph180,

# STATUS: APPROVED 😃 

git-merge

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**